ag-grid-webcomponent/index.js

```diff
@@ -68,72 +76,110 @@ class AgGrid extends HTMLElement {
   }
 
   static get observedAttributes() {
-    return observedProperties.map((property) => property.toLowerCase());
+    return agGrid.ComponentUtil.ALL_PROPERTIES.concat(setterProperties).map(
+      (property) => property.toLowerCase()
+    );
+  }
+
+  set columnState(state) {
+    this.columnApi.applyColumnState({ state: state, applyOrder: true });
   }
 
-  _isElmProperty(name) {
-    elmProperties.some((property) => property === name);
+  set disableResizeOnScroll(disabled) {
+    this._addEventHandler(
+      "onBodyScroll",
+      "disableResizeOnScroll",
+      function (params) {
+        if (!disabled) params.api.sizeColumnsToFit();
+      }
+    );
+  }
+
+  set filterState(state) {
+    this.api.setFilterModel(state);
+  }
+
+  set sizeToFitAfterFirstDataRendered(sizeToFit) {
+    this._addEventHandler(
+      "onFirstDataRendered",
+      "sizeToFitAfterFirstDataRendered",
+      function (params) {
+        if (sizeToFit) params.api.sizeColumnsToFit();
+      }
+    );
+  }
+
+  set rowData(data) {
+    this._applyChange("rowData", data);
+
+    if (this._agGrid.gridOptions.rowData === null) {
+      this.api.showNoRowsOverlay();
+    }
+  }
+
+  _applyChange(propertyName, newValue) {
+    let changeObject = {};
+    changeObject[propertyName] = { currentValue: newValue };
+
+    agGrid.ComponentUtil.processOnChange(
+      changeObject,
+      this._gridOptions,
+      this.api,
+      this.columnApi
+    );
+  }
+
+  _addEventHandler(eventName, type, callback) {
+    let collection = this._events[eventName] || {};
+    collection[type] = callback;
+
+    this._events = collection;
+
+    this._gridOptions[eventName] = function (args) {
+      Object.values(collection).map((event) => event(args));
+    };
   }
 
   attributeChangedCallback(name, oldValue, newValue) {
     // Re-Process GridConfig when an "Elm" property changed.
-    if (this._isElmProperty(name)) return this._process_grid_config();
-
     let gridPropertyName = this._propertyMap[name];
     let parsedNewValue = newValue ? JSON.parse(newValue) : newValue;
 
-    // for properties set before gridOptions is called
-    this._attributes[gridPropertyName] = parsedNewValue;
-
-    if (this._initialised) {
-      // for subsequent (post gridOptions) changes
-      let changeObject = {};
-      changeObject[gridPropertyName] = {
-        currentValue: parsedNewValue,
-      };
-
-      agGrid.ComponentUtil.processOnChange(
-        changeObject,
-        this._gridOptions,
-        this.api,
-        this.columnApi
-      );
+    if (this._definesSetter(gridPropertyName)) {
+      if (this._initialised) {
+        // for subsequent (post gridOptions) changes
+        this[gridPropertyName] = parsedNewValue;
+      } else {
```

Comment:
Do we really need to handle the pre init stuff in the `attributeChangedCallback` function? I feel like we can simplify this and make it easier to follow the flow.